src/stdlib_stringlist.f90

```diff
@@ -0,0 +1,996 @@
+contains
+
+! stringlist_index_add --
+!     Add an integer offset to the special index
+!
+! Arguments:
+!     index                 Special index
+!     offset                Offset to be added
+!
+function stringlist_index_add( index, offset )
+    type(stringlist_index_type), intent(in) :: index
+    integer, intent(in)                     :: offset
+
+    type(stringlist_index_type)             :: stringlist_index_add
+
+    stringlist_index_add        = index
+    stringlist_index_add%offset = stringlist_index_add%offset + offset
+end function stringlist_index_add
+
+! stringlist_index_subtract --
+!     Subtract an integer offset to the special index
+!
+! Arguments:
+!     index                 Special index
+!     offset                Offset to be subtracted
+!
+function stringlist_index_subtract( index, offset )
+    type(stringlist_index_type), intent(in) :: index
+    integer, intent(in)                     :: offset
+
+    type(stringlist_index_type)             :: stringlist_index_subtract
+
+    stringlist_index_subtract        = index
+    stringlist_index_subtract%offset = stringlist_index_subtract%offset - offset
+end function stringlist_index_subtract
+
+! compare string_type derived types
+!     Required by sorting functions
+!
+elemental logical function string_lower( string1, string2 )
+     type(string_type), intent(in) :: string1
+     type(string_type), intent(in) :: string2
+
+     string_lower = string1%value < string2%value
+end function string_lower
+
+elemental logical function string_greater( string1, string2 )
+     type(string_type), intent(in) :: string1
+     type(string_type), intent(in) :: string2
+
+     string_greater = string1%value > string2%value
+end function string_greater
+
+elemental logical function string_equal( string1, string2 )
+     type(string_type), intent(in) :: string1
+     type(string_type), intent(in) :: string2
+
+     string_equal = string1%value == string2%value
+end function string_equal
+
+function append_string( list, string )
+     type(stringlist_type), intent(in) :: list
+     character(len=*), intent(in)      :: string
+     type(stringlist_type)             :: append_string
+
+     append_string = list
+     call append_string%insert( list_after_end, string )
+end function append_string
+
+function prepend_string( string, list )
+     character(len=*), intent(in)      :: string
+     type(stringlist_type), intent(in) :: list
+     type(stringlist_type)             :: prepend_string
+
+     prepend_string = list
+     call prepend_string%insert( list_head, string )
+end function prepend_string
+
+function append_stringlist( slist, list )
+     type(stringlist_type), intent(in) :: list
+     type(stringlist_type), intent(in) :: slist
+     type(stringlist_type)             :: append_stringlist
+
+     append_stringlist = list
+     call append_stringlist%insert( list_after_end, slist )
+end function append_stringlist
+
+function append_stringarray( list, sarray )
```

Comment:
in my opinion it will be better to have arguments as ( sarray , list ) so that it becomes coherent with the previous functions convention (append_stringlist, prepend_string, etc.).